### ISSUE TITLE
ERROR Monitor ERROR:  node 2 does not belong to formation default

### ISSUE THREAD

**Issue by brsc2909** [citusdata/pg_auto_failover#456]
why am i getting this error when both my nodes are created with a new formation ? 

I create the first node and everything works fine
```
 /u01/pg/postgresql-12.3/bin/pg_autoctl create postgres  \
  --pgdata /u01/pg/pgcituscoordinator/data \
  --pghost localhost \
  --pgport 5432 \
  --listen '*' \
  --dbname postgres \
  --name pgsql801 \
  --hostname pgsql801 \
  --formation myformation \
  --monitor 'postgres://autoctl_node@pgmon801:5000/pg_auto_failover?sslmode=require' \
  --auth trust \
  --ssl-self-signed \
  --ssl-mode require
```

```
pg_autoctl show state  --formation myformation --pgdata /u01/pg/pgmonitorprod/data 
           Name |  Node |                                   Host:Port |       LSN | Reachable |       Current State |      Assigned State
----------------+-------+---------------------------------------------+-----------+-----------+---------------------+--------------------
pgsql801 |     2 | pgsql801:5432 | 0/160D5E8 |       yes |              single |              single
```

but when I create the seconds node using the same command it hangs and i getting these errors (journalctl -xe on first node)
```
Oct 14 20:10:51 pgsql801 postgres-pgcituscoordinator[17736]: 20:10:51 17739 ERROR Monitor ERROR:  node 2 does not belong to formation default
Oct 14 20:10:51 pgsql801 postgres-pgcituscoordinator[17736]: 20:10:51 17739 ERROR SQL query: SELECT * FROM pgautofailover.node_active($1, $2, $3, $4::pgautofailover.replication_state, $5, $6, $7)
Oct 14 20:10:51 pgsql801 postgres-pgcituscoordinator[17736]: 20:10:51 17739 ERROR SQL params: 'default', '2', '0', 'single', 'true', '0/17F6750', ''
Oct 14 20:10:51 pgsql801 postgres-pgcituscoordinator[17736]: 20:10:51 17739 ERROR Failed to get node state for node 2 in group 0 of formation "default" with initial state "single", replication state>
Oct 14 20:10:51 pgsql801 postgres-pgcituscoordinator[17736]: 20:10:51 17739 ERROR Failed to get the goal state from the monitor
Oct 14 20:10:51 pgsql801 postgres-pgcituscoordinator[17736]: 20:10:51 17739 WARN  Lost connection.
```

and i'm hanging with status:
```
pg_autoctl show state  --formation sriracha --group 0 --pgdata /u01/pg/pgmonitorprod/data  
           Name |  Node |                                   Host:Port |       LSN | Reachable |       Current State |      Assigned State
----------------+-------+---------------------------------------------+-----------+-----------+---------------------+--------------------
pgsql801 |     2 | pgsql801 :5432 | 0/160D5E8 |       yes |              single |              single
pgsql802 |     3 | pgsql802 :5432 |       0/0 |        no |        wait_standby |        wait_standby
```

Am I doing something wrong?

**Comment by DimCitus** [comment]
Hi @brsc2909 ; first question: which version of `pg_autoctl` are you using? the output header misalignment and the extra space at the right of the hostname and some other details are off, which make me wonder...

**Comment by brsc2909** [comment]
@DimCitus I'm using pg_auto_failover version 1.4.0 with postgres 12.3. Apologies the misalignment is because I added dummy host names instead of using the real ones.

I've managed to get it working by adding the `--run`to the create command but in practice i want to do the install using ansible so its not possible as the command does not exit. 

The scenario is like this
- run the `pg_autoctl create postgres...` without the `--run` on node 1
- start the service on node 1 using systemctl 
- run the `pg_autoctl create postgres...` without the `--run` on node 2
-  start the service on node 2 using systemctl 

This method worked fine for me using version 1.3.1, although i did have some random issues where the nodes would be added to seperate groups within the formation

**Comment by DimCitus** [comment]
Hey @brsc2909 ; can you see if https://github.com/neuroforgede/pg_auto_failover_ansible from @s4ke show you how to do it with ansible? Otherwise I would need to have more information to help debug your situation, starting with actual logs and output of commands, and the full logs if that's possible, from all the nodes involved.

**Comment by s4ke** [comment]
Since I got pinged: I'd be happy to get feedback on the playbooks that @DimCitus linked. But basically the process for setting up pg_autofailover completely automated can be found in the repository pretty nicely. If you want to play around with it, I'd suggest to check out the test suite in https://github.com/neuroforgede/pg_auto_failover_ansible/tree/master/test/simple which uses Vagrant to spin up a cluster.

**Comment by brsc2909** [comment]
Looks like this is fixed in #489

**Comment by DimCitus** [comment]
> Looks like this is fixed in #489

Thanks for the follow-up!
